xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -309,7 +309,7 @@ bool CAddonSettings::Load(const CXBMCTinyXML& doc)
     }
 
     // try to load the old setting value
-    if (!newSetting->FromString(setting.second))
+    if (newSetting && !newSetting->FromString(setting.second))
```

Comment:
Can a empty "newSetting" come in normal condition?
If no maybe to use:
```cpp
if (newSetting == nullptr || !newSetting->FromString(setting.second))
```
to have a log message about them?